src/package.ts

```diff
@@ -33,32 +33,29 @@ export function up(options?: find.Options): string | undefined {
  */
 export function cache(
 	name: string,
-	options?: find.Options & { create?: boolean },
+	options?: FindOptions & { create?: boolean },
 ): string | undefined {
-	options ||= {};
-
 	let dir = env.CACHE_DIR;
 
-	if (!dir || /^(1|0|true|false)$/.test(dir)) {
-		let pkg = up(options);
+	if (typeof dir === 'undefined' || dir === '1' || dir === '0' || dir === 'true' || dir === 'false') {
```

Comment:
This is wrong: `typeof dir === 'undefined` breaks on null and "" values. 

The rest is just pointless. Yes RegExp is technically slower than string matching, but this yields no **effective** gain. At best, an application would call this function a few times, a dozen at most. Expanding all this is just a waste of space so you can eek out (literally) zero noticeable difference for this function.

Reply:
I should change this to type !== string